tests/end_to_end/end_to_end.cpp

```diff
@@ -829,13 +834,181 @@ map_test(ebpf_execution_type_t execution_type)
     bpf_object__close(object);
 }
 
+bpf_sock_ops_t
+populate_bpf_sock_ops(
+    bpf_sock_op_type_t op,
+    uint8_t address_family,
+    const std::string& source_address,
+    uint16_t source_port,
+    const std::string& destination_address,
```

Comment:
Maybe at least add a // comment saying this function assumes source and dest address are globals since there is no interface or scope id parameter